addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/discovery/MinecraftDiscoveryService.java

```diff
@@ -0,0 +1,210 @@
+            public Observable<List<SignData>> call(ServerConnection connection) {
+                return connection.getSocketHandler().getSignsRx().distinct();
+            }
+        }, new Func2<ServerConnection, List<SignData>, Pair<ServerConnection, List<SignData>>>() {
+            @Override
+            public Pair<ServerConnection, List<SignData>> call(ServerConnection connection, List<SignData> signs) {
+                return new Pair<ServerConnection, List<SignData>>(connection, signs);
+            }
+        }).subscribe(new Action1<Pair<ServerConnection, List<SignData>>>() {
+            @Override
+            public void call(Pair<ServerConnection, List<SignData>> conectionSignPair) {
+                for (SignData sign : conectionSignPair.second) {
+                    submitSignDiscoveryResults(conectionSignPair.first.getThingUID(), sign);
+                }
+            }
+        }, new Action1<Throwable>() {
+            @Override
+            public void call(Throwable e) {
+                logger.error("Error while scanning for signs", e);
+            }
+        });
+
+        subscription.add(playerSubscription);
+        subscription.add(signSubscription);
+    }
+
+    /**
+     * Teardown subscribers and stop searching for players and signs.
+     */
+    private void stopDiscovery() {
+        if (subscription != null && !subscription.isUnsubscribed()) {
+            subscription.unsubscribe();
+        }
+    }
+
+    /**
+     * Get all servers that have been added as observable.
+     *
+     * @return observable emitting server objects.
+     */
+    private Observable<ServerConnection> serversConnectRx() {
+        return Observable.from(MinecraftHandlerFactory.getMinecraftServers())
+                .flatMap(new Func1<MinecraftServerHandler, Observable<ServerConnection>>() {
+                    @Override
+                    public Observable<ServerConnection> call(MinecraftServerHandler server) {
+                        ServerConfig config = server.getServerConfig();
+                        return ServerConnection.create(server.getThing().getUID(), config.getHostname(),
+                                config.getPort());
+                    }
+                });
+    }
+
+    /**
+     * Submit the discovered Devices to the Smarthome inbox,
+     *
+     * @param ip The Device IP
+     */
+    private void submitPlayerDiscoveryResults(ThingUID bridgeUID, String name) {
+
+        // uid must not contains dots
+        ThingUID uid = new ThingUID(MinecraftBindingConstants.THING_TYPE_PLAYER, bridgeUID.hashCode() + "_" + name);
+
+        if (uid != null) {
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(MinecraftBindingConstants.PARAMETER_PLAYER_NAME, name);
+            DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties).withBridge(bridgeUID)
+                    .withLabel("Minecraft Player (" + name + ")").build();
```

Comment:
I would simply put "name" as the label - the Thing type already says that this is a minecraft player.